matlab/doc/testing_guidelines_for_the_matlab_interface_to_apache_arrow.md

```diff
@@ -0,0 +1,97 @@
+```
+
+It might happen that caching or shadowing issue causes zero coverage because the test ends up executing some other source file. To avoid such problem, please first set a breakpoint in your source file and run the tests. This step is to make sure that your source file is executed by the tests. After this step, the command `runtests(___, "ReportCoverageFor", ___)` will work well.
+
+## Test File Location and Organization  
+
+All tests for MATLAB interface are put under `matlab/test`. To help easily locate related tests for each source file, test organization follows the following two rules:  
+
+- Test directory maps source directory. For example, the test directory `test/arrow/array/` contains all tests for the source directory `src/matlab/+arrow/+array/`.  
+- One test for one source file. For example, `test/arrow/array/tArray.m` is the test file for `src/matlab/+arrow/+array/Array.m`.  
+
+## When tests are run  
+
+When pushing changes to the `matlab` directory, GitHub Actions is used to automatically trigger tests under `matlab/test` to run. Currently, all tests under `matlab/test` are configured to run. In the future when the run time of running tests becomes a concern, we will consider doing source to test mapping and only triggering related tests to run.  
+
+## How to Write Tests  
+
+Tests for MATLAB interface should use the MATLAB testing framework [`matlab.unittest.TestCase class`](https://www.mathworks.com/help/matlab/ref/matlab.unittest.testcase-class.html). Here is a simple example.  
+
+```matlab
+classdef tArray < matlab.unittest.TestCase
+    methods(Test)
+        function InferNullsTrue(testCase)
+            matlabArray = [1 2 NaN 3];
+            arrowArray = arrow.array(matlabArray, InferNulls=true);
+            testCase.verifyEqual(arrowArray.Valid, [true; true; false; true]);
+        end
+    end
+end
+```
+
+More test examples can be found in the directory `matlab/test`.  
+
+## How to Run Tests Locally on Your Own Machine  
+
+### Prerequisites  
+
+The tests are written by using MATLAB testing framework [`matlab.unittest.TestCase class`](https://www.mathworks.com/help/matlab/ref/matlab.unittest.testcase-class.html). To run the tests on your machine, the following software must be installed on your machine:  
+
+1. [MATLAB](https://www.mathworks.com/products/get-matlab.html)
+2. [MATLAB Interface to Apache Arrow](https://github.com/mathworks/arrow/tree/main/matlab)  
+
+### Run tests in MATLAB  
+
+To run the MATLAB tests, start MATLAB and then cd to the test directory where test files of interest reside.  Call the `runtests` command to run your tests:  
+
+```matlab
+>> runtests(testFileName) % For example runtests("tArray.m")
+```
+
+To learn more about `runtests`, please check [the document of `runtests`](https://www.mathworks.com/help/matlab/ref/runtests.html).  
```

Comment:
```suggestion
To learn more about `runtests`, please check [the documentation for `runtests`](https://www.mathworks.com/help/matlab/ref/runtests.html).  
```

Reply:
I updated the example. Now there are two examples: one for running a single test file, and the other is for running tests under a test directory.